model/histogram/convert_test.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2025 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package histogram
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/prometheus/prometheus/model/labels"
+)
+
+type BucketExpectation struct {
+	le  string
+	val float64
+}
+
+type ExpectedHistogram struct {
+	buckets []BucketExpectation
+	count   float64
+	sum     float64
+}
+
+func TestConvertNHCBToClassicHistogram(t *testing.T) {
+	tests := []struct {
+		name      string
+		nhcb      any
+		labels    labels.Labels
+		expectErr bool
+		expected  ExpectedHistogram
+	}{
+		{
+			name: "Valid Histogram",
+			nhcb: &Histogram{
+				CustomValues:    []float64{1, 2, 3},
+				PositiveBuckets: []int64{10, 20, 30}, // Delta format: {10, 20, 30} -> Absolute: {10, 30, 60}
```

Comment:
Not sure if we need that comment here
```suggestion
				PositiveBuckets: []int64{10, 20, 30},
```